lib/administration/resource/src/resource_administration.cpp

```diff
@@ -87,51 +84,48 @@ namespace irods::experimental::administration::NAMESPACE_IMPL
         input.arg3 = _child.data();
 
         if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
-            return std::error_code{ec, std::generic_category()};
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(ec, fmt::format("Could not remove child resource [{}] from resource [{}]", _child, _parent));
         }
-
-        return {};
     }
 
-    auto resource_exists(RxComm& _comm, const resource_name_type _name)
-        -> std::tuple<std::error_code, bool>
+    auto resource_exists(RxComm& _comm, const resource_name_type _name) -> bool
     {
         try {
-            const auto gql = fmt::format("select RESC_ID where RESC_NAME = '{}'", _name);
-            return {std::error_code{}, query_builder{}.build(_comm, gql).size() > 0};
+            return query_builder{}.build(_comm, fmt::format("select RESC_ID where RESC_NAME = '{}'", _name)).size() > 0;
+        }
+        catch (const irods::exception& e) {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(e.code(), e.what());
```

Comment:
Do we need to do this? Looks like it is just re-constructing the exception and re-throwing without adding anything.

Also seen on 155 and 196.

Reply:
No. I'll adjust it.

Reply:
done